app.js

```diff
@@ -56,6 +60,29 @@ app.use(async (ctx, next) => {
     await next();
 });
 
+async function verifyAccountOwnership({ ctx, accountId, securityCode, signature }) {
```

Comment:
few nits:

1) it makes sense to structure this as Koa middleware, see e.g. https://github.com/nearprotocol/near-fiddle-api/blob/master/app.js#L88
2) to make it easier to reason about, it's best to make allowed cases explicit instead of making not allowed cases explicit.
3) Why use `securityCode` for smth that's not a security code anymore?

So whole method can be something like:

```
const currentBlock = (await ctx.near.connection.provider.status()).sync_info.latest_block_height;
if (givenBlock > currentBlock - 100 && givenBlock <= currentBlock) {
    const nearAccount = await ctx.near.account(accountId);
    if (!await verifySignature(nearAccount, securityCode, signature)) {
        ctx.throw(403);
    }
}
```

no need to handle all other errors as 403 as well, they already won't result in successful status